src/compound_merkle.rs

```diff
@@ -0,0 +1,264 @@
+use std::marker::PhantomData;
+
+use anyhow::{Context, Result};
+
+use crate::compound_proof::CompoundProof;
```

Comment:
BLOCKING: This name is really unfortunate because it clashes with the `rust-fil-proofs::storage_proofs` trait. Can we rename this, maybe to`CompoundMerkleProof`? Otherwise things may get very confusing when we try to integrate this with circuits (the next step).

Reply:
Good point, thanks.  I'll have it updated shortly.

Reply:
Thank you.